app/controllers/caterings_controller.rb

```diff
@@ -0,0 +1,46 @@
+class CateringsController < ApplicationController
+
+	def index
+		@caterings = Catering.all
+	end
+
+	def show
+		begin @catering = Catering.find(params[:id])
+		rescue ActiveRecord::RecordNotFound
+			flash.now.alert = "This Catering does not exist"
```

Comment:
Cant we use here our private 'redirect_not_found' function? Code looks like it doing the same thing as those function.